base/summarysize.jl

```diff
@@ -0,0 +1,148 @@
+immutable SummarySize
+    seen::ObjectIdDict
+    frontier_x::Vector{Any}
+    frontier_i::Vector{Int}
+    exclude::Any
+    chargeall::Any
+end
+
+
+"""
+    Base.summarysize(obj; exclude=Union{...}, chargeall=Union{...}) -> Int
+
+Compute the amount of memory used by all unique objects reachable from the argument.
+Keyword argument `exclude` specifies the types of objects to exclude from the traversal.
+Keyword argument `chargeall` specifies the types of objects to always charge the size of all of their fields, even if those fields would normally be excluded.
+"""
+function summarysize(obj::ANY; exclude::ANY = Union{DataType, TypeName}, chargeall::ANY = Union{TypeMapEntry, Method, Core.MethodInstance})
```

Comment:
Same comment regarding line lengths. You could just align `exclude` and `chargeall` beneath `obj`; that style is used elsewhere.